### PR TITLE
CGT-1389: Created allowable losses value input page

### PR DESCRIPTION
**Pull request by MJCallahanPage**

### CODE REVIEW

File: conf/res.routes [hunk 1]
```diff
@@ -41,20 +41,18 @@ POST    /other-properties                 controllers.resident.DeductionsControl
 
 #Allowable Losses Routes
 GET     /allowable-losses                 controllers.resident.DeductionsController.allowableLosses
+POST    /allowable-losses                 controllers.resident.DeductionsController.submitAllowableLosses
 
 #Allowable Losses Value Routes
 GET     /allowable-losses-value           controllers.resident.DeductionsController.allowableLossesValue
+POST    /allowable-losses-value           controllers.resident.DeductionsController.submitAllowableLossesValue
```
Comment (miranda-hawkes) [inline conf/res.routes]: Could you add a test in RoutesSpec for the submit please (and also for submitAllowableLosses if you fancy)

File: test/controllers/resident/DeductionsControllerTests/AllowableLossesValueActionSpec.scala [hunk 2]
```diff
@@ -21,25 +21,116 @@ import controllers.resident.DeductionsController
...
+    when(mockCalcConnector.fetchAndGetFormData[AllowableLossesValueModel]
+      (Matchers.eq(KeystoreKeys.ResidentKeys.allowableLossesValue))(Matchers.any(), Matchers.any()))
+      .thenReturn(Future.successful(getData))
+
+    new DeductionsController {
+      override val calcConnector: CalculatorConnector = mockCalcConnector
+    }
+  }
 
   "Calling .allowableLossesValue from the DeductionsController" should {
```
Comment (miranda-hawkes) [inline test/controllers/resident/DeductionsControllerTests/AllowableLossesValueActionSpec.scala]: Being picky, this 'should' should really be a 'when' now

File: test/controllers/resident/DeductionsControllerTests/AllowableLossesValueActionSpec.scala [hunk 3]
```diff
@@ -21,25 +21,116 @@ import controllers.resident.DeductionsController
...
-      status(result) shouldBe 200
+      "return a status of 200" in {
+        status(result) shouldBe 200
+      }
+
+      "return some html" in {
+        contentType(result) shouldBe Some("text/html")
+      }
+
+      "display the Reliefs Value view" in {
```
Comment (miranda-hawkes) [inline test/controllers/resident/DeductionsControllerTests/AllowableLossesValueActionSpec.scala]: should be Allowable Losses view

File: test/controllers/resident/DeductionsControllerTests/AllowableLossesValueActionSpec.scala [hunk 4]
```diff
@@ -21,25 +21,116 @@ import controllers.resident.DeductionsController
...
+    "an invalid form is submitted" should {
+      lazy val request = fakeRequestToPOSTWithSession(("amount", ""))
+      lazy val result = DeductionsController.submitAllowableLossesValue(request)
+      lazy val doc = Jsoup.parse(bodyOf(result))
+
+      "return a 400" in {
+        status(result) shouldBe 400
+      }
+
+      "render the Reliefs Value page" in {
```
Comment (miranda-hawkes) [inline test/controllers/resident/DeductionsControllerTests/AllowableLossesValueActionSpec.scala]: should be Allowable Losses view

**Comment by miranda-hawkes** [comment]
Please also fix merge conflicts :)

**Comment by MJCallahanPage** [comment]
Changes made and merge conflicts fixed

**Pull request merged**